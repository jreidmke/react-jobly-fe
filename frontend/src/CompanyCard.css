#company-logo {
    height: 3rem;
    width: 3rem;
    margin-right: 10px;
}

