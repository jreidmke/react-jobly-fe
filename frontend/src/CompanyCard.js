import './CompanyCard.css'
import { Link } from 'react-router-dom';

const CompanyCard = ({name, description, logo, handle}) => {
    const imgSrc = "https://www.brandcrowd.com/gallery/brands/pictures/picture15635752108148.jpg" 
    return(
        <div style={{width: '50%', margin: 'auto'}}>
            <span style={{display:'inline'}}>
                <h1><img src={imgSrc} alt='company-logo' id='company-logo'/><Link to={`/companies/${handle}`}>{name}</Link></h1>
            </span>
            <h3>{description}</h3>
            <hr></hr>
        </div>
    )
}

export default CompanyCard;