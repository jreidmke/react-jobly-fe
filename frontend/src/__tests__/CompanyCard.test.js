import { render } from '@testing-library/react';
import CompanyCard from '../CompanyCard';
import { MemoryRouter } from "react-router";

it("renders without crashing", function() {
    render(
        <MemoryRouter>
            <CompanyCard
                handle="pizza"
                name="Pizza School"
                description="Become an exceptional pizza maker in 16 weeks."
                logo_url="https://pbs.twimg.com/profile_images/770491761412173826/ZUeIa4tw_400x400.jpg"/>
        </MemoryRouter>
    )
});

it("matches snapshot", () => {
    const { asFragment } = render(
        <MemoryRouter>
            <CompanyCard
                handle="pizza"
                name="Pizza School"
                description="Become an exceptional pizza maker in 16 weeks."
                logo_url="https://pbs.twimg.com/profile_images/770491761412173826/ZUeIa4tw_400x400.jpg"/>
        </MemoryRouter>        
    );
    expect(asFragment()).toMatchSnapshot();
});