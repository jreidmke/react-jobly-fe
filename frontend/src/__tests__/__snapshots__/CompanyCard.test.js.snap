// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="width: 50%; margin: auto;"
  >
    <span
      style="display: inline;"
    >
      <h1>
        <img
          alt="company-logo"
          id="company-logo"
          src="https://www.brandcrowd.com/gallery/brands/pictures/picture15635752108148.jpg"
        />
        <a
          href="/companies/pizza"
        >
          Pizza School
        </a>
      </h1>
    </span>
    <h3>
      Become an exceptional pizza maker in 16 weeks.
    </h3>
    <hr />
  </div>
</DocumentFragment>
`;
